reinforcement_learning/code-env/python/wrapper/__init__.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+import gym
+import random
+
+import dataiku
+
+def train_agent(agent):
+    
+    # Get the parameters (the common ones)
+    environment_var = get_recipe_config()['environment']
+    agent_var = get_recipe_config()['agent']
+    policy_var = get_recipe_config()['policy']
+    gamma_var = get_recipe_config()['gamma']
+    lr_var = get_recipe_config()['dqn_learning_rate']
+    training_episodes_var = 5000
+
+    
+    # Create the JSON FILE and dump it into the output folder
+    training_infos = {
+        'name': environment_var,
+        'agent': agent_var,
+        'type': 'OpenAI Gym',
+        'num_episodes': training_episodes_var,
+        'lr': lr_var,
+        'gamma': gamma_var,
+        'policy': policy_var,
+        'training_date': str(datetime.datetime.now())
+    }
+    
+    saved_models = dataiku.Folder(get_output_names_for_role('main_output')[0])
+    saved_models_info = saved_models.get_info()
+    saved_models_path = saved_models.get_path()
+    
+    with open(saved_models_path + '/training_infos.json', 'w') as fp:
```

Comment:
Broken in MUS. I can't remember if there is a better way. Check the doc or ask Du.
But also, most of the time, you do not want libraries to write in specific places! Return the data you want to write and the recipe or the webapp will write. Writing file in the middle of logic makes it very rigid and impossible to unit test.